codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java

```diff
@@ -445,11 +436,12 @@ public String stringShape(StringShape shape) {
       ? "uint8"
       : "dafny.Char";
     var nilCheck = "";
+    String unAssertDataSource = dataSource.startsWith("input.(") ? "input" : dataSource;
```

Comment:
final

Reply:
Done here (including a lot of other stuffs) https://github.com/smithy-lang/smithy-dafny/commit/75c8e6a384e4c085e161578cfb682151c7d02e8e